clang/test/AST/ByteCode/constexpr-vectors.cpp

```diff
@@ -10,11 +10,229 @@ using FourI128VecSize __attribute__((vector_size(64))) = __int128;
+    b /= a;                                           \
+    b = (a += a);                                     \
+    b = (a -= a);                                     \
+    b = (a *= a);                                     \
+    b = (a /= a);                                     \
+    b = (b += a);                                     \
+    b = (b -= a);                                     \
+    b = (b *= a);                                     \
+    b = (b /= a);                                     \
+    a < b;                                            \
+    a > b;                                            \
+    a <= b;                                           \
+    a >= b;                                           \
+    a == b;                                           \
+    a != b;                                           \
+    a &&b;                                            \
+    a || b;                                           \
+    auto c = (a, b);                                  \
+    return c;                                         \
+  }
+
+// Ops specific to Integers.
+#define MathShiftOpsInts(Type)                            \
+  constexpr auto MathShiftopsInts##Type(Type a, Type b) { \
+    a = a << b;                                           \
+    a = a >> b;                                           \
+    a = a << 3;                                           \
+    a = a >> 3;                                           \
+    a = 3 << b;                                           \
+    a = 3 >> b;                                           \
+    a <<= b;                                              \
+    a >>= b;                                              \
+    a <<= 3;                                              \
+    a >>= 3;                                              \
+    b = (a <<= b);                                        \
+    b = (a >>= b);                                        \
+    b = (a <<= 3);                                        \
+    b = (a >>= 3);                                        \
+    a = a % b;                                            \
+    a &b;                                                 \
+    a | b;                                                \
+    a ^ b;                                                \
+    return a;                                             \
+  }
+
+MathShiftOps(FourCharsVecSize);
+MathShiftOps(FourIntsVecSize);
+MathShiftOps(FourLongLongsVecSize);
+MathShiftOps(FourFloatsVecSize);
+MathShiftOps(FourDoublesVecSize);
+MathShiftOps(FourCharsExtVec);
+MathShiftOps(FourIntsExtVec);
+MathShiftOps(FourLongLongsExtVec);
+MathShiftOps(FourFloatsExtVec);
+MathShiftOps(FourDoublesExtVec);
+
+MathShiftOpsInts(FourCharsVecSize);
+MathShiftOpsInts(FourIntsVecSize);
+MathShiftOpsInts(FourLongLongsVecSize);
+MathShiftOpsInts(FourCharsExtVec);
+MathShiftOpsInts(FourIntsExtVec);
+MathShiftOpsInts(FourLongLongsExtVec);
```

Comment:
Can you add tests for with `FourI128ExtVec` as well here? I think that might be broken.

Reply:
Sure, added more int128 tests, it's works well in my local.